judge/views/submission.py

```diff
@@ -531,6 +531,21 @@ def get(self, request, *args, **kwargs):
         return super(ForceContestMixin, self).get(request, *args, **kwargs)
 
 
+class UserAllContestSubmissions(ForceContestMixin, AllUserSubmissions):
+    def get_title(self):
+        return "%s's submissions in %s" % (self.username, self.contest.name)
+
+    def access_check(self, request):
+        super(UserAllContestSubmissions, self).access_check(request)
+        if not self.contest.users.filter(user_id=self.profile.id).exists():
```

Comment:
Not entirely sure of this but,
```suggestion
        if not self.contest.users.filter(user=self.profile).exists():
```

Reply:
Looks like this was copied from `UserContestSubmissions`. The queries generated are equivalent (and they are equally clear).